### PR TITLE
testcne: add functional tests for metrics lib to testcne

### PR DESCRIPTION
**Pull request by sushmasi**
Add functional tests relevant to usr/app/metrics/metrics.[ch]
to testcne.

Signed-off-by: Sushma Sitaram <sushma.sitaram@intel.com>

### CODE REVIEW

File: test/testcne/metrics_test.c [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+#include <metrics.h>         // for metrics_register
+#include <tst_info.h>        // for tst_error, tst_end, tst_start, TST_FAILED
+#include <getopt.h>          // for getopt_long, option
+
+#include "metrics_test.h"
+
+static int
+test_cb(metrics_client_t *client, const char *cmd __cne_unused, const char *params __cne_unused)
+{
+    printf("Test Callback\n");
```
Comment (jeffreybshaw) [inline test/testcne/metrics_test.c:17]: Can you please use cne_printf() instead of printf?

File: test/testcne/metrics_test.c [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+        goto err;
+    } else
+        tst_ok("PASS --- TEST: Added lport stats to metrics buffer\n");
+
+    ret = metrics_destroy();
+    if (ret < 0) {
+        tst_error("Unable to remove registred metrics commands\n");
+        goto err;
+    } else
+        tst_ok("PASS --- TEST: Removed all registred metrics commands\n");
```
Comment (jeffreybshaw) [inline test/testcne/metrics_test.c:73]: registered, not registred

**Review by jeffreybshaw — CHANGES_REQUESTED** [review]

File: test/testcne/metrics_test.c [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+    metrics_destroy();
+    free(client);
+    return -1;
+}
+
+int
+metrics_main(int argc, char **argv)
+{
+    tst_info_t *tst;
+    int verbose = 0, opt;
```
[the remark this answers is not in the document]
Comment (sushmasi) [reply test/testcne/metrics_test.c]: Removed

File: test/testcne/metrics_test.c (hunk 2 above)
Comment (sushmasi) [reply test/testcne/metrics_test.c]: Fixed typo

File: test/testcne/metrics_test.c (hunk 1 above)
Comment (sushmasi) [reply test/testcne/metrics_test.c]: Done

File: test/testcne/metrics_test.c [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+    metrics_client_t *client = calloc(1, sizeof(metrics_client_t));
+    client->cmd              = "/test";
+    client->params           = "params";
+    lport_stats_t stats      = {0};
+    char lpname[]            = "test_lport";
+
+    ret = metrics_init(NULL);
+    if (ret < 0) {
+        tst_error("Unable to initilize metrics library\n");
+        goto err;
```
Comment (elzamath) [inline test/testcne/metrics_test.c:35]: if we goto err; here , metrics_destroy() will be called. If metrics_init() fails, we don't want metrics_destroy() to be called, right?

File: test/testcne/metrics_test.c [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+static int
+metrics_test(void)
+{
+    const char *c;
+    int ret = 0;
+    metrics_client_t *client;
+    lport_stats_t stats = {0};
+    char lpname[]       = "test_lport";
+
+    client = calloc(1, sizeof(metrics_client_t));
```
Comment (jeffreybshaw) [inline test/testcne/metrics_test.c:28]: The conventional way of doing this is

```
client = calloc()
if (!client)
    return -1;

client->cmd = "/test"
```
This is a nit, but can you please change it?

File: test/testcne/metrics_test.c [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+        client->params = "params";
+    } else
+        return -1;
+
+    ret = metrics_init(NULL);
+    if (ret < 0) {
+        tst_error("Unable to initialize metrics library, %s\n", strerror(errno));
+        free(client);
+        return -1;
+    } else
```
Comment (jeffreybshaw) [inline test/testcne/metrics_test.c:40]: Having all the "} else" blocks here is not necessary. A simpler, more readable (IMO) implementation would do:

```
ret = metrics_init(NULL);
if (ret < 0) {
    tst_error("blah");
    free(client);
    return -1;
}

tst_ok("PASS --- TEST: Metrics library initialized\n");
```

**Review by jeffreybshaw — CHANGES_REQUESTED** [review]

File: test/testcne/metrics_test.c (hunk 6 above)
Comment (sushmasi) [reply test/testcne/metrics_test.c]: Done

File: test/testcne/metrics_test.c (hunk 5 above)
Comment (sushmasi) [reply test/testcne/metrics_test.c:28]: Done

**Review by jeffreybshaw — APPROVED** [review]

File: test/testcne/meson.build [hunk 7]
```diff
@@ -122,6 +124,7 @@ test_names = [
     'log',
     'mbuf',
     'mempool',
+    'metrics',
```
[the remark this answers is not in the document]
Comment (jeffreybshaw) [reply test/testcne/meson.build]: We should use this. I will add a change to our unit-test application to make it possible to mark a test as "skipped" when the test determines it cannot properly run.

**Pull request merged**